security/pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,147 @@
+	Tag tag.Key
+
+	// A TagValue represents a Tag with a specific value. It is used to record
+	// values for a Metric.
+	TagValue tag.Mutator
+)
+
+// Value creates a new TagValue for the Tag.
+func (t Tag) Value(value string) TagValue {
+	return tag.Upsert(tag.Key(t), value)
+}
+
+// MustCreateTag will attempt to create a new Tag. If
+// creation fails, then this method will panic.
+func MustCreateTag(key string) Tag {
+	k, err := tag.NewKey(key)
+	if err != nil {
+		panic(fmt.Errorf("could not create tag key %q: %v", key, err))
+	}
+	return Tag(k)
+}
+
+// MustRegisterViews is a helper function that will ensure that the views created
+// for the provided Metrics are registered. If the view for a metric fails to register,
+// this method will panic.
+func MustRegisterViews(metrics ...Metric) {
+	for _, s := range metrics {
+		if err := view.Register(s.View()); err != nil {
+			panic(err)
+		}
+	}
+}
+
+// NewSum creates a new Metric with an aggregation type of Sum. That means that data collected
+// by the new Metric will be summed before export.
+func NewSum(name, description string, tags ...Tag) Metric {
+	return newMetric(name, description, view.Sum(), tags...)
+}
+
+// NewGauge creates a new Metric with an aggregation type of LastValue. That means that data collected
+// by the new Metric will export only the last recorded value.
+func NewGauge(name, description string, tags ...Tag) Metric {
+	return newMetric(name, description, view.LastValue(), tags...)
+}
+
+// NewDistribution creates a new Metric with an aggregration type of Distribution. This means that the
+// data collected by the Metric will be collected and exported as a histogram, with the specified bounds.
+func NewDistribution(name, description string, bounds []float64, tags ...Tag) Metric {
+	return newMetric(name, description, view.Distribution(bounds...), tags...)
+}
+
+func newMetric(name, description string, aggregation *view.Aggregation, tags ...Tag) Metric {
+	return newFloat64Metric(name, description, aggregation, tags...)
+}
+
+type float64Metric struct {
+	*stats.Float64Measure
+
+	tags []tag.Mutator
+	view *view.View
+}
+
+func newFloat64Metric(name, description string, aggregation *view.Aggregation, tags ...Tag) *float64Metric {
+	measure := stats.Float64(name, description, stats.UnitDimensionless)
+	tagKeys := make([]tag.Key, 0, len(tags))
+	for _, t := range tags {
+		tagKeys = append(tagKeys, tag.Key(t))
+	}
+	return &float64Metric{
+		measure,
+		make([]tag.Mutator, 0, len(tags)),
+		&view.View{Measure: measure, TagKeys: tagKeys, Aggregation: aggregation},
+	}
+}
+
+func (f *float64Metric) Increment() {
+	f.Record(1)
+}
+
+func (f *float64Metric) Name() string {
+	return f.Float64Measure.Name()
+}
+
+func (f *float64Metric) Record(value float64) {
+	stats.RecordWithTags(context.Background(), f.tags, f.M(value)) //nolint:errcheck
+}
+
+func (f *float64Metric) With(tagValues ...TagValue) Metric {
+	t := make([]tag.Mutator, 0, len(f.tags)+len(tagValues))
```

Comment:
note: this is actually a bug in the original code. this should be something like:

`t := make([]tag.Mutator, len(f.tags))`

Reply:
Fixed this in both places for now, but we should get this file hoisted up sooner rather than later to prevent any accidental divergence down the road.